cmake/parthenonConfig.cmake.in

```diff
@@ -0,0 +1,36 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# Favor using the kokkos package that was built with parthenon, if one has not been specified
+if (@PARTHENON_EXTERNAL_KOKKOS@)
+  find_package(Kokkos 3 REQUIRED PATHS @Kokkos_DIR@ NO_DEFAULT_PATH)
+else()
+  find_package(Kokkos 3 REQUIRED PATHS @CMAKE_INSTALL_PREFIX@ NO_DEFAULT_PATH)
+endif()
+
+set(Kokkos_BUILT_WITH_CUDA @Kokkos_ENABLE_CUDA@)
+if(${Kokkos_BUILT_WITH_CUDA})
+  string(FIND ${CMAKE_CXX_COMPILER} "nvcc_wrapper" FOUND_WRAPPER)
+  if( ${FOUND_WRAPPER} EQUAL "-1" )
+    if (NOT @PARTHENON_EXTERNAL_KOKKOS@)
+      message(WARNING "Kokkos was built with cuda, recommend setting CMAKE_CXX_COMPILER to @CMAKE_INSTALL_PREFIX@/bin/nvcc_wrapper")
+    else()
+      message(WARNING "Kokkos was built with cuda, recommend setting CMAKE_CXX_COMPILER to @PARTHENON_EXTERNAL_KOKKOS@/bin/nvcc_wrapper")
```

Comment:
I thought PARTHENON_EXTERNAL_KOKKOS was a switch, did you mean to use Kokkos_DIR?

Reply:
Gah, you're right, thanks for catching this (I'll re-run my test cases once this latest round of revisions settles down)

Reply:
Once this PR went in, we can make another PR that add the example for the documentation and have the CI build that after the install step.

Reply:
> Once this PR went in, we can make another PR that add the example for the documentation and have the CI build that after the install step.

What do you mean? The documentation is already added in the Building.md file? What is left to do?

Reply:
Add an `examples` folder with the pi example and a `CMakeLists.txt` like:
```cmake
cmake_minimum_required(VERSION 3.11)
project(parthenon_linking_example)
if(NOT TARGET Parthenon::parthenon)
  find_package(parthenon REQUIRED)
endif()
add_executable(
  pi-example
  pi_driver.cpp
  pi_driver.hpp
  calculate_pi.cpp
  calculate_pi.hpp
  )
target_link_libraries(pi-example PRIVATE Parthenon::parthenon)
```
This `CMakeLists.txt` will work for internal as well as external builds, so you can build that example as part of parthenon, but also standalone. The latter you can add to the CI, so that the exports are always tested.

Reply:
Oh, ok I see, I was thinking some kind of test would be a good idea. I will create an issue.